### ISSUE TITLE
module-import-not-at-top-of-file (E402) should not warn about logging configuration

### ISSUE THREAD

**Issue by JimDabell** [astral-sh/ruff#19928]
### Summary

It’s frequently necessary to configure logging before importing a package that logs. For instance, if a package `foo`  logs `DEBUG` level events, you might want to write:

```python
import logging


logging.basicConfig(level=logging.DEBUG)


import foo
```

[The documentation notes:](https://docs.astral.sh/ruff/rules/module-import-not-at-top-of-file/)

> This rule makes an exception for both `sys.path` modifications (allowing for `sys.path.insert`, `sys.path.append`, etc.) and `os.environ` modifications between imports.

I believe this exception should also extend to logging configuration.

**Comment by ntBre** [comment]
It's not quite a duplicate, but we're having a similar discussion on E402 in #19904. If we take a more general approach there, we could fold this issue in as well. `logging` probably fits better into the existing special cases since it's also from the stdlib, though.
